### DISCUSSION TITLE
Matataki、DAPP、Fan票 的生态系统架构

### DISCUSSION THREAD

**Discussion by Linklinkguan** [Matataki-io/matataki-frontend#1006] [Discussion]
目前在DAPP中提供的功能都比较缺乏交互反馈，不同系统之间的联系比较散乱。比如在Matataki Quest中有人申请完成了你的自定义任务，但是你并不能通过系统内的渠道接受到这条消息，因为DAPP本身没有开发通知系统，即便有通知，如果用户不登入Quest也是无法查看的。主要问题是在系统架构上没有做到互联共通，反而是早就了各种信息的孤岛。

一些优化的思路
1. 邮件通知系统，只要是使用Matataki账号登录的DAPP或是Matataki本体中产生的消息，全部会经由邮件通知。
2. 在Fan票主页中做聚合器，将在DApp中与该Fan票相关的信息都反向聚合到Fan票主页中，并进行分类展示
3. ……

**Comment by Linklinkguan** [comment]
其实把本体的提醒理解成一个收件箱也不会奇怪就是。Mastodon上收到其他实例的提醒也是一例，Inbox/Outbox+订阅发布模型。可以理解为@someone@matataki.io 收到了 @gm@quest.matataki.io的消息

**Reply by whyouare111** [reply]
做成收件箱形式也方便暴露API转RSS，和各种bot集成都方便，订阅到Mastodon或者Telegram都可以，汇集到用户最常用的信息源

**Reply by whyouare111** [reply]
概念上的落脚点类似这样： 1. matataki提供一个收件箱服务；2. matataki的收件箱支持订阅生态内提供的各种订阅源； 3. matataki官方会提供各种APP，包括写作和确权平台，任务等等，这些APP都有对应频道，有使用就作为默认频道订阅,表现形式可以是Matataki上的助手/bot账号的通知；4. 在UI上现在的Matataki本体比较适合展现这个提醒，所以目前把提醒入口放在这里; 5. 也提供API可以自行制作bot通过其他方式订阅，如RSS,ActivityPub,Telegram

**Reply by Linklinkguan** [reply]
—— by 夜桜NEKO

> 根據分散式應用程式白皮書理論 (The General Theory of Decentralized Applications White Paper)[3]，符合以下標準的申請則被視爲分散式應用程式：
> 
> 開源 (Open Source) - 分散式應用程式必須完全開源並自主運行，簡單來説是將程式碼完全公開，以便其他開發者或程式人員作復核及修改。[4]
> 分散 (Decentralization) - 將所有數據加密存放於可公開訪問的區塊鏈中。[5]
> 密碼貨幣 (Tokens) - 為區塊鏈提供原生的數字貨幣 (Coins) 或用智能合約發表的代幣 (Token) ，為采礦人員和用戶提供價值。
> 算法 (Algorithm) - 使用標準加密算法通過節點網絡啟用采礦。[6]

指 Dapp通知应该独立，Dapp creator 和 Source creator 的关系只是合作关系，并不存在提供和兼容通知系统的义务，对应的则需要 Matataki 提供足够的 API。
